cindy/index.js

```diff
@@ -0,0 +1,197 @@
+// isEqualSlices(8, 3, 2) ➞ true
+// isEqualSlices(8, 3, 3) ➞ false
+// isEqualSlices(24, 12, 2) ➞ true
+function isEqualSlices(slices, person, perPerson) {
+	return slices >= person * perPerson;
+}
+console.log(isEqualSlices(24, 12, 2))
+
+//11. XO Create a function that takes a string, checks if it has the same number of 'x's and 'o's and returns either true or false.
+// Notes:
+// Return a boolean value (true or false).
+// The string can contain any character.
+// When neither an x nor an o is in the string, return true.
+// Examples:
+// isEqualNumXandO("ooxx") ➞ true
+// isEqualNumXandO("xooxx") ➞ false
+// isEqualNumXandO("ooxXm") ➞ true (case insensitive)
+// isEqualNumXandO("zpzpzpp") ➞ true (returns true if no x and o)
+// isEqualNumXandO("zzoo") ➞ false
+function isEqualNumXandO(str) {
+  str = str.toLowerCase();
+  let arrayOfCharacters = str.split("");
+  let countX = arrayOfCharacters.reduce( function( n, val ) {
+      return n + (val === 'x');
+    }, 0);
+  let countO = arrayOfCharacters.reduce( function( n, val ) {
+      return n + (val === 'o');
+    }, 0);
+  if ( countX == countO ) {
+    return true;
+  } else {
+    return false;
+  }
+}
+console.log(isEqualNumXandO("ooxx"))
+console.log(isEqualNumXandO("xooxx"))
+console.log(isEqualNumXandO("ooxXm"))
+
+//12. isPrime? Create a function that returns true if a number is prime and false if it's not. NB: a prime number is any positive integer greater than 1, which is only evenly divisible by two divisors: itself and 1. The first ten prime numbers are 2, 3, 5, 7, 11, 13, 17, 19, 23 and 29.
+// Examples:
+// isPrime(7) ➞ true
+// isPrime(9) ➞ false
+// isPrime(10) ➞ false
+function isPrime(num) {
+  for(let i = 2; i < num; i++)
+    if(num % i === 0) return false;
+  return num > 1;
+}
+console.log("isPrime? => ", isPrime(10))
+
+//13. Validate Email. Create a function that takes a string, checks for valid email address syntax, and then returns either true or false accordingly.
+
+// NOTE: email syntax is more complicated than you might think (see https://en.wikipedia.org/wiki/Email_address#Syntax), so we do not expect you to cover all cases! Just work your way through the below simplified validation criteria:
+
+// The string must contain exactly one "@" character.
+// The string must contain at least one "." character.
+// The "." and the "@" must be in the appropriate places.
+// e.g. "." cannot be the last character.
+// e.g. "." cannot be directly before or after "@"
+// e.g. there cannot be consecutive "." characters
+// e.g. "@" must have at least one character in front of it.
+// e.g. "j@example.com" is valid while "@example.com" is invalid.
+// e.g. "john.smith@com" is invalid while "john.smith@email.com" is valid.
+// e.g. "john..smith@email.com" and "john.@email.com" and "john@.email.com" are all invalid.
+function validateEmail(email) 
+    {
+        var re = /\S+@\S+\.\S+/;
+        return re.test(email);
+    }
+    
```

Comment:
Task 13 ==> "john..smith@email.com" and "john.@email.com" and "john@.email.com" should be all invalid(false), but you face true! The correct way to solve this task is to use if , else if statements

Reply:
ok done it! (but it's not my solution :/ )

Reply:
Thanks:))